src/main/java/bio/terra/cda/app/service/QueryService.java

```diff
@@ -52,51 +67,66 @@ private Job runJob(Job queryJob) {
     return queryJob;
   }
 
-  // For now, hardcode the known list of systems. In the future, we will get this from the
-  // database itself, as each published dataset will have a list of contributing systems.
-  private enum Source {
-    GDC,
-    PDC;
-
-    final String match;
-
-    Source() {
-      match = String.format("\"system\": \"%s\"", name());
-    }
-
-    boolean match(String s) {
-      return s.contains(match);
+  /**
+   * Convert a BQ value to a json node.
+   *
+   * @param value the value to convert
+   * @param field the schema field for the value
+   * @return a json node for the value
+   */
+  @VisibleForTesting
+  protected JsonNode valueToJson(FieldValue value, Field field) {
```

Comment:
I think it's possible to `value` to be null so you may want to add a check for that

Reply:
Actually `null` is handled by a `value` whose attribute is `PRIMITIVE` and `isNull()` returns `true`.

https://github.com/CancerDataAggregator/cda-service/blob/ps-issue-67-duplicate-json-keys/src/main/java/bio/terra/cda/app/service/QueryService.java#L96

If this check isn't done, then you get an NPE when calling `getStringValue()` or `getValue()`